### PR TITLE
[v1.4] Select command tests

### PR DESCRIPTION
**Pull request by adamwth**
commits after 7af6c7b are what's new on this PR

- add tests for SelectCommand (both unit test and system test)
- implemented Deselect feature for data removal commands (i.e. checkout, clear, undo, redo)

### CODE REVIEW

**Comment by adamwth** [comment]
ready for review

File: src/test/data/XmlUtilTest/validRoom.xml [hunk 1]
```diff
@@ -10,8 +10,8 @@
             <tagged>vip</tagged>
         </guest>
         <bookingPeriod>
-            <startDate>29/10/2018</startDate>
-            <endDate>4/11/2018</endDate>
+            <startDate>30/10/2018</startDate>
```
Comment (pikulet) [inline src/test/data/XmlUtilTest/validRoom.xml]: Same as earlier, should not have to push this file every PR o.o

File: src/main/java/seedu/address/commons/util/XmlUtil.java [hunk 2]
```diff
@@ -68,4 +69,27 @@
         m.marshal(data, file.toFile());
     }
 
+    /**
+     * Saves the data in the file in xml format.
```
Comment (pikulet) [inline src/main/java/seedu/address/commons/util/XmlUtil.java]: Javadoc can explain the difference between `saveDataToFile` and `saveDataToFileIfMissing`

File: src/main/java/seedu/address/logic/commands/ClearCommand.java [hunk 3]
```diff
@@ -18,6 +20,8 @@
     public CommandResult execute(Model model, CommandHistory history) {
         requireNonNull(model);
         model.resetData(getEmptyConcierge());
+        EventsCenter.getInstance().post(new DeselectGuestListEvent());
```
Comment (pikulet) [inline src/main/java/seedu/address/logic/commands/ClearCommand.java]: What if a room list is currently being selected? Will the same effect be achieved?

File: src/main/java/seedu/address/commons/events/ui/DeselectRoomListEvent.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
+package seedu.address.commons.events.ui;
+
+import seedu.address.commons.events.BaseEvent;
+
+/**
+ * Represents a deselect room list event
+ */
+public class DeselectRoomListEvent extends BaseEvent {
```
Comment (pikulet) [inline src/main/java/seedu/address/commons/events/ui/DeselectRoomListEvent.java]: In what cases would this event be used? A room is never actually "deleted" from Concierge after all.

Comment (adamwth) [reply src/main/java/seedu/address/commons/events/ui/DeselectRoomListEvent.java]: yup you're right. I forgot to remove it after coming to the same conclusion just now.

File: src/test/data/XmlUtilTest/validRoom.xml (hunk 1 above)
Comment (adamwth) [reply src/test/data/XmlUtilTest/validRoom.xml]: the saveDataToFileIfMissing is supposed to fix this. see if it works tomorrow

File: src/main/java/seedu/address/commons/events/ui/DeselectRoomListEvent.java (hunk 4 above)
Comment (adamwth) [reply src/main/java/seedu/address/commons/events/ui/DeselectRoomListEvent.java]: fixed in latest commit

File: src/main/java/seedu/address/commons/util/XmlUtil.java (hunk 2 above)
Comment (adamwth) [reply src/main/java/seedu/address/commons/util/XmlUtil.java]: fixed in latest commit

**Pull request merged**